apps/www/components/ui/navigation-menu.tsx

```diff
@@ -0,0 +1,122 @@
+import React from "react"
+import * as NavigationMenuPrimitive from "@radix-ui/react-navigation-menu"
+import { cva, type VariantProps } from "class-variance-authority"
+import { ChevronDown } from "lucide-react"
+
+import { cn } from "@/lib/utils"
+
+export const NavigationMenuRoot: React.FC<
+  NavigationMenuPrimitive.NavigationMenuProps
+> = ({ className, ...props }) => (
+  <NavigationMenuPrimitive.Root
+    className={cn("relative z-[1] flex flex-1 justify-center", className)}
+    {...props}
+  />
+)
+
+export const NavigationMenuList: React.FC<
+  NavigationMenuPrimitive.NavigationMenuListProps
+> = ({ className, ...props }) => (
+  <NavigationMenuPrimitive.List
+    className={cn(
+      "group flex flex-1 list-none justify-center rounded-lg border border-slate-300 bg-white p-1 dark:border-slate-700 dark:bg-slate-800",
+      className
+    )}
+    {...props}
+  />
+)
+
+export const NavigationMenuItem = NavigationMenuPrimitive.NavigationMenuItem
+
+const navigationMenuTriggerStyle = cva(
+  "rounded-md py-2 px-3 font-medium leading-none hover:bg-slate-100 focus:shadow-sm focus:ring-2 focus:ring-slate-300 focus:bg-slate-100 dark:focus:bg-slate-700 dark:hover:bg-slate-700 dark:focus:ring-slate-500 outline-none w-full transition-all duration-300"
+)
+
+export const NavigationMenuTrigger: React.FC<
+  NavigationMenuPrimitive.NavigationMenuTriggerProps
+> = ({ className, children, ...props }) => (
+  <NavigationMenuPrimitive.Trigger
+    className={cn(
+      navigationMenuTriggerStyle(),
+      "group flex items-center justify-center gap-0.5",
+      className
+    )}
+    {...props}
+  >
+    {children}{" "}
+    <ChevronDown
+      className="relative top-[1px] h-4 w-4 transition duration-200 group-data-[state=open]:rotate-180"
+      aria-hidden
+    />
+  </NavigationMenuPrimitive.Trigger>
+)
+
+const navigationMenuLinkVariants = cva([], {
```

Comment:
Yeah. We can skip variants and keep the styles inline. The consumers of the component can adjust the styles as needed.

Reply:
I use this variant to change the style of the NavigationMenuLink when it's inside a NavigationMenuContent like ListItem component in demo